### PR TITLE
Fix HalfLine creation and its Boundary object

### PR DESCRIPTION
**Pull request by mathbagu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
N/A

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Refactoring

**What is the current behavior?** *(You can also link to an open issue here)*
The implementation of the TieLineAdder and the HalfLineBoundary is not really good

**What is the new behavior (if this is a feature change)?**
This PR propose an alternative implementation that seems better to me.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineAdderImpl.java [hunk 1]
```diff
@@ -124,6 +124,11 @@ public TieLineAdderImpl add() {
             return TieLineAdderImpl.this;
         }
 
+        public TieLineImpl.HalfLineImpl build() {
```
Comment (MioRtia) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineAdderImpl.java:127]: Can this be package-private?

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineImpl.java [hunk 2]
```diff
@@ -7,30 +7,47 @@
...
-        boolean fictitious = false;
-        double r = Double.NaN;
-        double x = Double.NaN;
-        double g1 = Double.NaN;
-        double g2 = Double.NaN;
-        double b1 = Double.NaN;
-        double b2 = Double.NaN;
-        HalfLineBoundaryImpl boundary;
+
+        public HalfLineImpl(String id, String name, boolean fictitious, double r, double x, double g1, double b1, double g2, double b2, Branch.Side side) {
```
Comment (MioRtia) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineImpl.java:35]: Can this be package-private?

**Review by MioRtia — APPROVED** [review]
Two small remarks but otherwise okay for me!

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineAdderImpl.java (hunk 1 above)
Comment (mathbagu) [reply iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineAdderImpl.java:127]: I change the visibility to private: this method should be called only by the enclosing TieLineAdder instance

**Pull request merged**